platform/background_downloader_ios.mm

```diff
@@ -197,9 +197,13 @@ - (void)clear {
 #pragma mark - NSURLSessionDownloadDelegate implementation
 
 - (void)finishDownloading:(NSURLSessionTask *)downloadTask error:(nullable NSError *)error {
-  [self.restoredTasks removeObjectForKey:downloadTask.currentRequest.URL.path];
+  if (error)
+    NSLog(@"BackgroundDownloadTask error: %@ ", error);
 
-  TaskInfo *info = [self.tasks objectForKey:@(downloadTask.taskIdentifier)];
+  if (downloadTask.currentRequest)
```

Comment:
А в чём конкретно была ошибка? Разве не нужно убирать таск в любом случае, независимо от ошибки, как было раньше?

Reply:
downloadTask.currentRequest = nil, and next removeObjectForKey produces exception

Reply:
Please split this PR into two.
To properly fix ios downloader, please try the original code and replace currentRequest with originalRequest. The issue here is either the request was started incorrectly (with nil instead of proper Request instance) or the server response changed the originalRequest to nil by some reason.

Reply:
currentRequest == originalRequest == nil

Reply:
How to reproduce it? It means an "empty" request. This is not a normal situation.

Reply:
Guys, it would be nice to release the new version this weekend. Could we split this PR into two parts and merge the new data ASAP?